entities/src/main/java/org/odk/collect/entities/javarosa/filter/LocalEntitiesFilterStrategy.kt

```diff
@@ -87,7 +87,13 @@ class LocalEntitiesFilterStrategy(entitiesRepository: EntitiesRepository) :
 
         return if (candidate != null) {
             val child = candidate.nodeSide.steps[0].name.name
-            val value = candidate.evalContextSide(sourceInstance, evaluationContext) as String
+            var value = candidate.evalContextSide(sourceInstance, evaluationContext)
```

Comment:
Where does the type of `value` come from? Could it also be any other type like maybe a date type or boolean?

Reply:
In the filter it is just a number like `age=25` but JR keeps numbers as doubles so it is `25.0`. Before converting it to string we need to get rid of that `.0` which I described above.

Reply:
> Could it also be any other type like maybe

Yes I was thinking about dates as well. I'm going to check that.

Reply:
Great, thanks. My recollection is that this was addressed for the simple expression case. I would have expected the same tests related to that to catch this. Is there branching between the simple expression and complex expression (with and/or) case? Am I remembering incorrectly about those tests?

Reply:
> My recollection is that this was addressed for the simple expression case. I would have expected the same tests related to that to catch this. Is there branching between the simple expression and complex expression (with and/or) case? Am I remembering incorrectly about those tests?

It's not clear to me what you're referring to here @lognaturel. What's the simple expression case?

Reply:
I distinctly remember a conversation you and I had about doubles when you were first writing support for expressions without and/or. I thought this class of problems was addressed there but I see that the form crashes in v2024.3.x too so maybe not?

Reply:
> I distinctly remember a conversation you and I had about doubles when you were first writing support for expressions without and/or. I thought this class of problems was addressed there but I see that the form crashes in v2024.3.x too so maybe not?

Yeah, I don't remember considering it at this level sadly!